### ISSUE TITLE
Common Lines: fuzzy mask & Gaussian filter

### ISSUE THREAD

**Issue by j-c-c** [ComputationalCryoEM/ASPIRE-Python#861]
Many of the common-lines methods in the MATLAB code apply a `fuzzy_mask` to the projections images and a `Gaussian_filter` to the polar Fourier images prior to estimating orientations. This is not implemented in here in the Python code.

**Comment by janden** [comment]
Both of these should help (fuzzy mask reduces influence of noise in the corners which we know is not informative and Gaussian filtering makes shift estimation slightly more likely to succeed). Would be worth to implement this at some point (can be done as extra preprocessing steps).

**Comment by j-c-c** [comment]
Turns out we are already using a Gaussian filter for the common-lines methods. It was "hiding" here in the `_generate_shift_phase_and_filter()` [method](https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/82763cf928ce4b921e7f5b7a4389d7b13f3ef246/src/aspire/abinitio/commonline_base.py#L424-L449), line 447.

**Comment by garrettwrong** [comment]
Yeah, I thought we talked about it, but I didn't want to jump the gun. Cool, thanks.  If it's just the fuzzy mask, I think that is implemented as a method in ASPIRE already.  Optimistically could just optionally enable. 📿

**Comment by j-c-c** [comment]
Tested out the `CLSyncVoting` with and without a "`fuzzy_mask`ed" source for a couple different levels of SNR. Saw pretty good improvement when using the fuzzy mask. I will open a PR and add a fuzzy mask option to the common line methods. Might be good to use the mask by default. 

Here are the results comparing estimated rotations to ground truth ("mad" is mean_angular_distance in degrees):

For SNR = 0.5:
    unmasked mse = 3.18, masked mse = 2.55
    unmasked mad = 75, masked mad = 65

For SNR = 0.75:
    unmasked mse = 2.78, masked mse = 0.44
    unmasked mad = 67, masked mad = 20

Here is the script I used:
[fuzzy_script.zip](https://github.com/ComputationalCryoEM/ASPIRE-Python/files/12499478/fuzzy_script.zip)
